sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1658,19 +1658,14 @@ class Analyzer(
       // Third, for every Window Spec, we add a Window operator and set currentChild as the
       // child of it.
       var currentChild = child
-      var i = 0
-      while (i < groupedWindowExpressions.size) {
-        val ((partitionSpec, orderSpec), windowExpressions) = groupedWindowExpressions(i)
+      groupedWindowExpressions.foreach { case ((partitionSpec, orderSpec), windowExpressions) =>
```

Comment:
Hi @srowen , It seems `groupedWindowExpressions`  is `Seq[((Seq[Expression], Seq[SortOrder]), ArrayBuffer[NamedExpression])]` but the desired output is `LogicalPlan`. So, it seems I can directly use `fold` or `reduce` simply. So, could this just use `foreach` rather than others? Otherwise, please teach me for a better expression.